### DISCUSSION TITLE
rego_parse_error when using "every" keyword.

### DISCUSSION THREAD

**Discussion by robhafner** [open-policy-agent/community#546] [OPA and Rego] [answered]
I'm attempting to use the "every" keyword as described in the documentation found [here](https://www.openpolicyagent.org/docs/latest/#for-all-every). However, I'm receiving the following rego parse error when I do. 

1 error occurred: policy.rego:21: rego_parse_error: unexpected identifier token: missing body
	group != "ProhibitedGroup"
	

Any help on what I may be doing incorrectly would be appreciated! Thanks in advance!

https://play.openpolicyagent.org/p/pLXIx1FC0m

**Comment by srenatus** [comment, marked as answer]
Your playground example read
```rego
deny if {
	every group in claims.authorities
	group != "ProhibitedGroup"
}
```

Try adding `{ ... }`:

```rego
deny if {
	every group in claims.authorities {
		group != "ProhibitedGroup"
	}
}
```

`every` only applies to the expressing within its `{ ... }` block. That is because it's not ever valid to do something like this:
```rego
# invalid
deny contains group if {
	every group in claims.authorities
	group != "ProhibitedGroup"
}
```
Since `every` is true when the domain (`claims.authorities`) is empty, we wouldn't have any `group` to use, although the expression is true.


Note that you can drop the outermost `{ .. }`, and go with this here:
```rego
deny if every group in claims.authorities {
	group != "ProhibitedGroup"
}
```

**Reply by robhafner** [reply]
Thanks for the quick response and explanation! The braces resolved the issue and your explanation helped me understand the syntax further. Much appreciated!
